server/src/endpoints/qs/push_notification_provider.rs

```diff
@@ -6,30 +6,75 @@ use async_trait::async_trait;
 use jsonwebtoken::{encode, Algorithm, EncodingKey, Header};
 use phnxbackend::{
     qs::{PushNotificationError, PushNotificationProvider},
-    settings::ApnsSettings,
+    settings::{ApnsSettings, FcmSettings},
 };
 use phnxtypes::messages::push_token::{PushToken, PushTokenOperator};
 use reqwest::{Client, StatusCode};
 use serde::{Deserialize, Serialize};
+use serde_json::{json, Value};
 use std::{
     fs::File,
     io::Read,
-    sync::{Arc, Mutex},
+    sync::Arc,
     time::{SystemTime, UNIX_EPOCH},
 };
+use tokio::sync::Mutex;
+
+#[derive(Debug, Serialize)]
+struct FcmClaims {
+    iss: String,
+    scope: String,
+    aud: String,
+    iat: usize,
+    exp: usize,
+}
+
+// Struct for the Google OAuth2 response
+#[derive(Debug, Deserialize)]
+struct OauthTokenResponse {
+    access_token: String,
+    _token_type: String,
+    expires_in: u64,
+}
 
 #[derive(Debug, Serialize, Deserialize)]
-struct Claims {
+struct ApnsClaims {
     iss: String,
     iat: usize,
 }
 
 #[derive(Debug, Clone)]
-pub struct ApnsToken {
+struct ApnsToken {
     jwt: String,
     issued_at: u64,
 }
 
+#[derive(Debug, Clone)]
+struct FcmToken {
+    token: String,
```

Comment:
minor: we could also zeroize this token. Not sure it is needed though.

Reply:
I added it to the whole struct